stabilizer/src/main/java/com/hazelcast/stabilizer/coordinator/Coordinator.java

```diff
@@ -372,6 +374,30 @@ private void echo(String msg) {
         log.info(msg);
     }
 
+    private void uploadResourcesToAgents() throws IOException {
+        if (!new File(UPLOAD_HOME).exists()) {
+            log.info("Skipping upload, since no upload file in working directory");
+            return;
+        }
+        log.info("UPLOAD HOME:" + UPLOAD_HOME);
+        List<File> files = Utils.getFilesFromClassPath(UPLOAD_HOME);
+        for (String ip : agentsClient.getPublicAddresses()){
+            for (File file : files){
+                String syncCommand = format("rsync -avv -e \"ssh %s\" %s %s@%s:hazelcast-stabilizer-%s/workers/%s/",
+                        props.get("SSH_OPTIONS", ""),
+                        file ,
+                        props.get("USER"),
+                        ip,
+                        getVersion(),
+                        testSuite.id);
+                bash.execute(syncCommand);
+            }
+            log.info("    " + ip + " copied");
+        }
+        log.info(format("Finished copying resources file '%s' to agents", UPLOAD_HOME));
```

Comment:
Since the user calls this directory 'upload' don't confuse him with 'resources'. It is very important to use the same word, else you are forcing a translation table onto the user and this means that he finds the product difficult to understand. You are also not copying a file, but potentially many files.

So probably best to say:

log.info(format("Finished copying '+%s+' directory agents", UPLOAD_DIR));